source/particles/particle_handler.cc

```diff
@@ -1222,140 +1205,141 @@ namespace Particles
     const typename Triangulation<dim, spacedim>::CellStatus         status,
     const boost::iterator_range<std::vector<char>::const_iterator> &data_range)
   {
-    // -------------------
-    // HOTFIX: static_cast std::vector to void*
-    // TODO: Apply ParticleHandler to variable size transfer
-    //    const void *data = static_cast<const void *>(data_vector.data());
-    const void *data = static_cast<const void *>(&*(data_range.begin()));
-    // -------------------
-
-    const unsigned int *n_particles_in_cell_ptr =
-      static_cast<const unsigned int *>(data);
-    const void *pdata =
-      reinterpret_cast<const void *>(n_particles_in_cell_ptr + 1);
-
-    if (*n_particles_in_cell_ptr == 0)
-      return;
+    std::vector<Particle<dim, spacedim>> loaded_particles_on_cell =
```

Comment:
I had been wondering why you didn't make this variable `const` until I realized that you (cleverly!) use `std::move` when building the multimap again. Maybe add a comment here to this effect.